pkg/analytics/client.go

```diff
@@ -0,0 +1,232 @@
+
+	var ID string
+	if err := d.Get(GAClientIDKey, &ID); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			// couldn't find existing Id, so generate one and store it
+			ID = newClientID()
+			logrus.Debugf("Didn't find existing clientID, generating a new one: %s", ID)
+			_ = d.Put(GAClientIDKey, ID)
+
+			return ID, nil
+		}
+		// unknown error getting key
+		return newClientID(), nil
+	}
+
+	return ID, nil
+}
+
+// Generates a new unique id
+func newClientID() string {
+	return uuid.New().String()
+}
+
+// enableGoogleAnalytics enable google analytics for HPE CLI
+func enableGoogleAnalytics() bool {
+	var disableGA string
+
+	d, err := db.Open()
+
+	if err != nil {
+		logrus.Debug("Unable to open DB to get DisableAnalyticsKey")
+
+		return false
+	}
+	defer d.Close()
+
+	if err := d.Get(DisableAnalyticsKey, &disableGA); err != nil {
+		return true
+	}
+
+	if disableGA == "true" {
+		logrus.Debugf("Found existing DisableAnalyticsKey, updating it's value to : %s", disableGA)
+
+		err := d.Put(DisableAnalyticsKey, "false")
+		if err != nil {
+			logrus.Debugf("Unable to put the key %s in to DB", DisableAnalyticsKey)
+			return false
+		}
+
+		return true
+	} else if disableGA == falseValue {
+		return true
+	}
+
+	return false
+}
+
+// disableGoogleAnalytics disable google analytics for HPE CLI
+func disableGoogleAnalytics() bool {
+	var disableGA string
+
+	d, err := db.Open()
+
+	if err != nil {
+		logrus.Debug("Unable to open DB to get GAClientIDKey ")
+		return false
+	}
+	defer d.Close()
+
+	if err := d.Get(DisableAnalyticsKey, &disableGA); err != nil {
+		logrus.Debugf("Didn't find existing DisableAnalyticsKey, creating a new one: %s", DisableAnalyticsKey)
+
+		err := d.Put(DisableAnalyticsKey, "true")
+		if err != nil {
+			logrus.Debugf("Unable to put the key %s in to DB", DisableAnalyticsKey)
+			return false
+		}
+
+		return true
+	}
+
+	if disableGA == falseValue {
+		logrus.Debugf("Didn't find existing enableAnalytics, creating a new one: %s", DisableAnalyticsKey)
+		_ = d.Put(DisableAnalyticsKey, "true")
+
+		return true
+	} else if disableGA == "true" {
+		return true
+	}
+
+	return false
+}
+
+// CheckGoogleAnalytics Check whether google analytics is enabled or disabled in db
+func CheckGoogleAnalytics() bool {
+	var disableGA string
+
+	d, err := db.Open()
+
+	if err != nil {
+		logrus.Debug("Unable to open DB to get GAClientIDKey ")
+		return false
+	}
+	defer d.Close()
+
+	if err := d.Get(DisableAnalyticsKey, &disableGA); err != nil {
+		logrus.Debugf("Didn't found existing DisableAnalyticsKey key %s", disableGA)
+
+		return true
+	}
+
+	if disableGA == falseValue {
+		return true
+	}
+
+	return false
+}
+
+// TrackEvent Measurement Protocol api to track user events
+func (c *Client) TrackEvent() (string, error) {
+	const uriPath = "/collect"
+
+	const host = "https://www.google-analytics.com"
+
+	u, err := url.Parse("https://www.google-analytics.com/collect")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	u.Scheme = "https"
```

Comment:
This would be redundant.
The above parse statement should populate all of these values based on the Parse function.
These statements are then unnecessary